shoreman.sh

```diff
@@ -74,16 +74,23 @@ load_env_file() {
 # ## Reading the Procfile
 
 # The Procfile needs to be parsed to extract the process names and commands.
-# The file is given on stdin, see the `<` at the end of this while loop.
 run_procfile() {
   local procfile=${1:-'Procfile'}
-  while read line || [[ -n "$line" ]]; do
-    if [[ -z "$line" ]] || [[ "$line" == \#* ]]; then continue; fi
+  local IFS=$'\n'
+  local lines=( $(grep "^[A-Za-z0-9_-]\{1,\}:[[:space:]]*.\{1,\}" "$procfile") )
+  local line=''
+
+  for line in "${lines[@]}"; do
```

Comment:
@shamrin from the bash manual:
```
       ${!name[@]}
       ${!name[*]}
              If name is an array variable,  expands  to  the  list  of  array
              indices  (keys)  assigned  in  name.   If  name is not an array,
              expands to 0 if name is set and null otherwise.  When @ is  used
              and the expansion appears within double quotes, each key expands
              to a separate word.
```
hope this is what PR author tried to achieve